### ISSUE TITLE
Rotation via setNativeProps transform doesn't center

### ISSUE THREAD

**Issue by riglar** [react-native-community/react-native-svg#815]
Great library. I'm trying to work with 7.1.2 on iOS (no expo) to animate the rotation of a circle. The circle is definitely moving, but seems to be rotating around 0,0 - I need it to rotate around the center point (24, 24). Looks like originX/Y aren't being applied in this scenario. 

The code I'm currently using is below. Is this a way I can achieve center point rotation?

Happy to write a PR if the native code needs extending to support this.

Many thanks
Tom

react-native-svg@7.1.2
react-native@0.57.1
xcode@10.0

```
import * as React from 'react'
import Svg, { Path, Circle } from 'react-native-svg'
import { Animated } from 'react-native'

const radius = 22.8
const circumference = radius * 2 * Math.PI
const offset = circumference - (40 / 100) * circumference

export class MicrophoneActiveIcon extends React.Component {
  public circleRef: any

  public state = {
    rotation: new Animated.Value(-90)
  }

  public componentDidMount() {
    Animated.loop(
      Animated.timing(this.state.rotation, {
        toValue: 270,
        duration: 1000,
        useNativeDriver: true
      })
    ).start()

    this.state.rotation.addListener(({ value }) => {
      this.circleRef.setNativeProps({
        transform: [{ rotate: value.toString() }]
      })
    })
  }

  public render() {
    return (
      <Svg width="48" height="48">
        <Circle
          ref={ref => {
            this.circleRef = ref
          }}
          cx="24"
          cy="24"
          r={radius.toString()}
          stroke="black"
          strokeWidth={2}
          fill="none"
          originX="24"
          originY="24"
          strokeLinecap="round"
          strokeDasharray={`${circumference.toString()} ${circumference.toString()}`}
          strokeDashoffset={offset.toString()}
        />
      </Svg>
    )
  }
}

```

**Comment by msand** [comment]
Can you try removing the object with the transform property which is wrapping the object with the rotation, i think then it should apply the origin props as well.

**Comment by msand** [comment]
Hmm, actually, that probably wouldn't work either, as the origin properties wouldn't be given in setNativeProps, perhaps something more like if you want to use react-native transforms:

```jsx
    this.state.rotation.addListener(({ value }) => {
      this.circleRef.setNativeProps({
        transform: [
          { translateX: 24 },
          { translateY: 24 },
          { rotate: value.toString() },
          { translateX: -24 },
          { translateY: -24 },
        ]
      })
    })
  }
```
or this if you want the legacy react-native-svg shorthands
```jsx
    this.state.rotation.addListener(({ value }) => {
      this.circleRef.setNativeProps({
          rotate: value,
          originX: 24,
          originY: 24
      })
    })
  }
```
or this if you want standard svg
```jsx
    this.state.rotation.addListener(({ value }) => {
      this.circleRef.setNativeProps({
          transform: `rotate(${value}, 24, 24)`
      })
    })
  }
```

Or, alternatively, just use the viewBox attribute to move the origin to the center of the canvas and set cx/cy to zero:
```jsx
  public render() {
    return (
      <Svg width="48" height="48" viewBox="-24 -24 48 48">
        <Circle
          ref={ref => {
            this.circleRef = ref
          }}
          cx="0"
          cy="0"
          r={radius.toString()}
          stroke="black"
          strokeWidth={2}
          fill="none"
          strokeLinecap="round"
          strokeDasharray={`${circumference.toString()} ${circumference.toString()}`}
          strokeDashoffset={offset.toString()}
        />
      </Svg>
    )
  }
```

**Comment by msand** [comment]
Btw, it doesn't really make sense to use the native driver + a javascript event listener to call setNativeProps, if you're using setNativeProps to drive the changes you probably shouldn't use the native driver. Either wrap the Circle with animated and set the style={{transform}} there (for plain native driven animation), or remove/set useNativeDriver to false.

**Comment by riglar** [comment]
Thanks for your detailed response @msand

I've tried all your suggestions and seem to be getting decent behaviour using this transform:
```
 { translateX: 24 },
 { translateY: 24 },
 { rotate: value.toString() },
 { translateX: -24 },
 { translateY: -24 },
```

I've unset useNativeDriver as you suggest.

Favour owed, hit me up if you need PR work doing.

**Comment by riglar** [comment]
@msand, my implemented solution doesn't work in 8.0.8.

Would you expected that? Or has something gone wrong.

**Comment by msand** [comment]
Hmm, i would expect it work, or don't know of any related changes at least
